### PR TITLE
Carver mappings

### PR DESCRIPTION
**Pull request by jaskarth**
Maps some of the new carver changes in 21w08a, as well as a few surrounding mappings.

### CODE REVIEW

File: mappings/net/minecraft/world/gen/carver/Carver.mapping [hunk 1]
```diff
@@ -9,20 +10,61 @@ CLASS net/minecraft/class_2939 net/minecraft/world/gen/carver/Carver
...
+		ARG 3 chunk
+		ARG 4 posToBiome
+		ARG 5 seed
+		ARG 18 carvingMask
+		ARG 19 skipFunction
+	METHOD method_33979 getState (Lnet/minecraft/class_5871;)Lnet/minecraft/class_2680;
+		ARG 0 config
+	METHOD method_33980 isDebug (Lnet/minecraft/class_5871;)Z
+		ARG 0 config
+	CLASS class_5874 SkipFunction
```
Comment (liach) [inline mappings/net/minecraft/world/gen/carver/Carver.mapping]:
```suggestion
	CLASS class_5874 SkipPredicate
```
Just based on my instinct that this returns a boolean

File: mappings/net/minecraft/world/gen/carver/Carver.mapping [hunk 2]
```diff
@@ -9,20 +10,61 @@ CLASS net/minecraft/class_2939 net/minecraft/world/gen/carver/Carver
...
+	METHOD method_33977 isRegionUncarvable (Lnet/minecraft/class_2791;IIIIII)Z
+		ARG 1 chunk
+	METHOD method_33978 carveRegion (Lnet/minecraft/class_5873;Lnet/minecraft/class_5871;Lnet/minecraft/class_2791;Ljava/util/function/Function;JIDDDDDLjava/util/BitSet;Lnet/minecraft/class_2939$class_5874;)Z
+		ARG 1 context
+		ARG 2 config
+		ARG 3 chunk
+		ARG 4 posToBiome
+		ARG 5 seed
+		ARG 18 carvingMask
+		ARG 19 skipFunction
```
Comment (liach) [inline mappings/net/minecraft/world/gen/carver/Carver.mapping]:
```suggestion
		ARG 19 skipPredicate
```

File: mappings/net/minecraft/world/gen/carver/CaveCarver.mapping [hunk 3]
```diff
@@ -1,16 +1,38 @@
...
+		ARG 4 posToBiome
+		ARG 5 seed
+		ARG 7 seaLevel
 		ARG 8 x
 		ARG 10 y
 		ARG 12 z
 		ARG 14 yaw
 		ARG 15 yawPitchRatio
 		ARG 17 carvingMask
+		ARG 18 skipFunction
```
Comment (liach) [inline mappings/net/minecraft/world/gen/carver/CaveCarver.mapping]:
```suggestion
		ARG 18 skipPredicate
```

File: mappings/net/minecraft/world/gen/carver/CaveCarver.mapping [hunk 4]
```diff
@@ -1,16 +1,38 @@
...
 	METHOD method_12675 carveTunnels (Lnet/minecraft/class_5873;Lnet/minecraft/class_5871;Lnet/minecraft/class_2791;Ljava/util/function/Function;JIDDDDDFFFIIDLjava/util/BitSet;Lnet/minecraft/class_2939$class_5874;)V
-		ARG 7 mainChunkZ
+		ARG 1 contex
```
Comment (liach) [inline mappings/net/minecraft/world/gen/carver/CaveCarver.mapping]:
```suggestion
		ARG 1 context
```
this method is cringe (26 slots)! Thank you for your efforts

**Comment by liach** [comment]
Need an update to merge into 08b.

**Review by liach — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Pull request merged**